crates/rpc/rpc-types/src/eth/trace/tracerequest.rs

```diff
@@ -0,0 +1,62 @@
+//! Builder style functions for `trace_call`
+
+use crate::{state::StateOverride, trace::parity::TraceType, BlockOverrides, CallRequest};
+use reth_primitives::BlockId;
+use std::collections::HashSet;
+
+use serde::{Deserialize, Serialize};
+
+/// Trace Request builder style function implementation
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TraceRequest {
+    /// call request object
+    pub call: CallRequest,
+    /// trace types
+    pub trace_types: HashSet<TraceType>,
+    block_id: Option<BlockId>,
+    state_overrides: Option<StateOverride>,
+    block_overrides: Option<Box<BlockOverrides>>,
+}
+
+impl TraceRequest {
+    /// Returns a new [`TraceRequest`] given a [`CallRequest`] and [`HashSet<TraceType>`]
+    pub fn new(call: CallRequest, trace_types: HashSet<TraceType>) -> Self {
+        Self { call, trace_types, block_id: None, state_overrides: None, block_overrides: None }
+    }
+
```

Comment:
we also need functions that

1. insert a single trace type
2. insert multiple that accepts impl IntoIter
3. then 3 more helper functions that call 1. with the tracetype variants